src/Microsoft.DotNet.Interactive.Http.Tests/HttpKernelTests.cs

```diff
@@ -111,6 +111,158 @@ public async Task can_handle_multiple_request_in_a_single_submission()
         requests.Select(r => r.RequestUri.AbsoluteUri).ToArray().Should().BeEquivalentTo(new[] { "https://location1.com:1200/endpoint", "https://location2.com:1200/endpoint" });
     }
 
+    [Fact]
+    public async Task variables_in_submit_request_are_saved()
```

Comment:
The simplest way to verify the variable is to use the `RequestValue` command rather than via a second `SubmitCode`. This approach should make the test simpler and easier to understand.